src/App.tsx

```diff
@@ -358,6 +373,51 @@ function App() {
     setInvokeMethodResults({});
   };
 
+  /**
+   * setup caip294:wallet_announce event listener
+   * docs: https://github.com/ChainAgnostic/CAIPs/blob/bc4942857a8e04593ed92f7dc66653577a1c4435/CAIPs/caip-294.md#specification
+   */
+  useEffect(() => {
+    const handleWalletAnnounce = (ev: Event) => {
+      const customEvent = ev as CustomEvent;
+      const newExtensionId = customEvent.detail.params.extensionId ?? '';
```

Comment:
Seems like this is expecting pre-CAIP-341 format for the `wallet_announce` event. Are we thinking we just merge it and change it later?